docker/puppetserver/docker-entrypoint.d/80-ca.sh

```diff
@@ -43,13 +43,15 @@ else
   hocon -f /etc/puppetlabs/puppetserver/conf.d/ca.conf \
     set certificate-authority.allow-subject-alt-names "${CA_ALLOW_SUBJECT_ALT_NAMES}"
 
-  # Generate the same SSL directory that the PE installer creates.
-  #
-  # The steps in this file correspond to the steps in the PE installer.
-  #
-  # See https://github.com/puppetlabs/puppet-enterprise-modules/blob/kearney/modules/pe_install/manifests/prepare/certificates.pp
+  [ -d "$CADIR" ] && [ -f "$CADIR/ca_crt.pem" ]
```

Comment:
I don't think the file test will ever pass if the directory isn't there, so do we need the `-d`?

Reply:
Also, I think these variables have slightly confusing names since they're named opposite of what I would expect. Maybe do something like this to flip the exit code:

```shell
new_ca_exists=$([ -f "$CADIR/ca_crt.pem" ] && echo 1)
old_ca_exists=$([ -f "$SSLDIR/ca/ca_crt.pem" ] && echo 1)
```

So that then you can use the the `-n` (not null) and `-z` (null) tests in shell code like:

```shell
# no existing CA
if [ -z "$new_ca_exists" ] && [ -z "$old_ca_exists" ]; then
# legacy CA upgrade
elif [ -z "$new_ca_exists" ] && [ -n "$old_ca_exists" ]; then
fi
```